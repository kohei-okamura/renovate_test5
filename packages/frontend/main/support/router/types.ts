/*
 * Copyright Â© 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { Route } from 'vue-router'

export type RouteParams = Route['params']

export type RouteQuery = Route['query']
