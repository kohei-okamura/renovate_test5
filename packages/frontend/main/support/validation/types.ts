/*
 * Copyright Â© 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { ValidationObserver, ValidationProvider } from 'vee-validate'
import { DateLike } from '~/models/date'
import { CustomRuleParams } from '~/support/validation/rules/custom'

type BetweenRule = {
  min: number
  max: number
}

type BooleanRule = boolean

type CustomRule = CustomRuleParams

type MinDateRule =
  (DateLike | undefined) |
  [DateLike | undefined] |
  [DateLike | undefined, string] |
  {
    minValue: DateLike | undefined
    attribute?: string
  }

type LengthRule = number | {
  len: number
  max?: number
}

type NumericRule = number | string

type NonItemsZeroRule = {
  itemName: string
}

export type Rule = {
  asciiAlphaNum?: BooleanRule
  alphaNum?: BooleanRule
  between?: BetweenRule
  custom?: CustomRule
  digits?: NumericRule
  email?: BooleanRule
  fax?: BooleanRule
  hiragana?: BooleanRule
  length?: LengthRule
  katakana?: BooleanRule
  max?: NumericRule
  min?: NumericRule
  minDate?: MinDateRule
  minValue?: NumericRule
  nonItemsZero?: NonItemsZeroRule
  integer?: BooleanRule
  numeric?: BooleanRule
  postcode?: BooleanRule
  required?: BooleanRule
  tel?: BooleanRule
  validTimeDuration?: BooleanRule
  zenginDataRecordChar?: BooleanRule
}

export type Rules = {
  [key: string]: Rule | Rule[] | Rules | Rules[] | undefined
}

export type ValidationObserverInstance = InstanceType<typeof ValidationObserver>

export type ValidationProviderInstance = InstanceType<typeof ValidationProvider>
