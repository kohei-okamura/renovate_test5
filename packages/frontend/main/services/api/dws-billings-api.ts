/*
 * Copyright © 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { DwsBillingStatus } from '@zinger/enums/lib/dws-billing-status'
import { AxiosInstance } from 'axios'
import { DwsBilling, DwsBillingId } from '~/models/dws-billing'
import { DwsBillingBundle } from '~/models/dws-billing-bundle'
import { DwsBillingCopayCoordination } from '~/models/dws-billing-copay-coordination'
import { DwsBillingServiceReport } from '~/models/dws-billing-service-report'
import { DwsBillingStatement } from '~/models/dws-billing-statement'
import { Job } from '~/models/job'
import { api, Api } from '~/services/api/core'

/**
 * 障害福祉サービス：請求 API.
 */
export namespace DwsBillingsApi {

  export type CopyParams = {
    id: DwsBillingId
  }

  export type CopyResponse = {
    job: Job
  }

  export type Copy = {
    copy (params: CopyParams): Promise<CopyResponse>
  }

  export type CreateForm = {
    officeId?: number
    transactedIn?: string
    providedIn?: string
  }
  export type CreateParams = {
    form: CreateForm
  }
  export type CreateResponse = {
    job: Job
  }
  export type Create = {
    create (params: CreateParams): Promise<CreateResponse>
  }

  export type FileParams = Api.Id & Api.Token
  export type FileResponse = Api.Url
  export type File = {
    file (params: FileParams): Promise<FileResponse>
  }

  export type GetParams = Api.GetParams
  export type GetResponse = {
    billing: DwsBilling
    bundles: DwsBillingBundle[]
    statements: DwsBillingStatement[]
    reports: DwsBillingServiceReport[]
    copayCoordinations: DwsBillingCopayCoordination[]
  }

  export type GetIndexParams = Api.GetIndexParams & {
    officeId?: number | ''
    statuses?: DwsBillingStatus[] | ''
    start?: string
    end?: string
  }
  export type GetIndexResponse = Api.GetIndexResponse<DwsBilling>

  export type UpdateStatusForm = {
    status: DwsBillingStatus
  }
  export type UpdateStatusParams = Api.Id & {
    form: UpdateStatusForm
  }
  export type UpdateStatusResponse = GetResponse & {
    job?: Job
  }
  export type UpdateStatus = {
    updateStatus (params: UpdateStatusParams): Promise<UpdateStatusResponse>
  }

  export type Definition =
    Copy &
    Create &
    Api.Get<GetResponse> &
    Api.GetIndex<GetIndexParams, GetIndexResponse> &
    File &
    UpdateStatus

  const endpoint = (id?: DwsBillingId, ...segments: string[]) => api.endpoint('dws-billings', id, ...segments)

  export const create = (axios: AxiosInstance): Definition => ({
    async copy ({ id }) {
      return await api.extract(axios.post(api.endpoint('dws-billings', id, 'copy')))
    },
    async create ({ form }) {
      return await api.extract(axios.post(endpoint(), form))
    },
    async file ({ id, token }) {
      return await api.extract(axios.get(endpoint(id, 'files', token)))
    },
    async get ({ id }) {
      return await api.extract(axios.get(endpoint(id)))
    },
    async getIndex (params) {
      return await api.extract(axios.get(endpoint(), { params }))
    },
    async updateStatus ({ id, form }) {
      return await api.extract(axios.put(endpoint(id, 'status'), form))
    }
  })
}
