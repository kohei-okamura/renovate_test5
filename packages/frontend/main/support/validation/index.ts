/* eslint-disable camelcase */
/*
 * Copyright © 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { configure, extend, localize, ValidationObserver, ValidationProvider } from 'vee-validate'
import {
  alpha_num as alphaNum,
  between,
  digits,
  email,
  integer,
  length,
  max,
  max_value as maxValue,
  min,
  min_value as minValue,
  numeric,
  required
} from 'vee-validate/dist/rules'
import { VueConstructor } from 'vue'
import { asciiAlphaNum } from '~/support/validation/rules/ascii-alpha-num'
import { custom } from '~/support/validation/rules/custom'
import { fax } from '~/support/validation/rules/fax'
import { hiragana } from '~/support/validation/rules/hiragana'
import { katakana } from '~/support/validation/rules/katakana'
import { minDate } from '~/support/validation/rules/min-date'
import { nonItemsZero } from '~/support/validation/rules/non-items-zero'
import { postcode } from '~/support/validation/rules/postcode'
import { tel } from '~/support/validation/rules/tel'
import { validTimeDuration } from '~/support/validation/rules/valid-time-durations'
import { zenginDataRecordChar } from '~/support/validation/rules/zengin-data-record-char'

function setupComponents (vue: VueConstructor): void {
  vue.component('ValidationObserver', ValidationObserver)
  vue.component('ValidationProvider', ValidationProvider)
}

function setupRules (): void {
  extend('alphaNum', alphaNum)
  extend('asciiAlphaNum', asciiAlphaNum)
  extend('between', between)
  extend('custom', custom)
  extend('digits', digits)
  extend('email', email)
  extend('fax', fax)
  extend('hiragana', hiragana)
  extend('integer', integer)
  extend('katakana', katakana)
  extend('length', length)
  extend('max', max)
  extend('maxValue', maxValue)
  extend('min', min)
  extend('minDate', minDate)
  extend('minValue', minValue)
  extend('nonItemsZero', nonItemsZero)
  extend('numeric', numeric)
  extend('postcode', postcode)
  extend('required', required)
  extend('tel', tel)
  extend('validTimeDuration', validTimeDuration)
  extend('zenginDataRecordChar', zenginDataRecordChar)
}

function setupMessages (): void {
  localize('ja', {
    messages: {
      alphaNum: '半角英数字で入力してください',
      between: (_, params = {}) => `${params.min}以上、${params.max}以下の半角数字で入力してください。`,
      digits: (_, params = {}) => `${params.length}桁の半角数字で入力してください。`,
      email: '有効なメールアドレスを入力してください。',
      integer: '半角数字のみで入力してください。',
      length: (_, params = {}) => `${params.length}文字で入力してください。`,
      max: (_, params = {}) => `${params.length}文字以内で入力してください。`,
      maxValue: (_, params = {}) => `${params.max}以下を入力してください。`,
      min: (_, params = {}) => `${params.length}文字以上で入力してください。`,
      minValue: (_, params = {}) => `${params.min}以上を入力してください。`,
      numeric: '半角数字のみで入力してください。',
      required: '入力してください。'
    }
  })
}

/**
 * VeeValidate 関連のセットアップ処理.
 */
export function setupVeeValidate (vue: VueConstructor): void {
  configure({
    mode: 'eager'
  })
  setupComponents(vue)
  setupRules()
  setupMessages()
}
