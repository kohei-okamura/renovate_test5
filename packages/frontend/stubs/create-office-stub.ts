/*
 * Copyright © 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { OfficeQualification } from '@zinger/enums/lib/office-qualification'
import { OfficeStatus } from '@zinger/enums/lib/office-status'
import { Purpose } from '@zinger/enums/lib/purpose'
import { range } from '@zinger/helpers'
import toKatakana from 'jaco/fn/toKatakana'
import { Office, OfficeId } from '~/models/office'
import { DWS_AREA_GRADE_ID_MAX, DWS_AREA_GRADE_ID_MIN } from '~~/stubs/create-dws-area-grade-stub'
import { LTCS_AREA_GRADE_ID_MAX, LTCS_AREA_GRADE_ID_MIN } from '~~/stubs/create-ltcs-area-grade-stub'
import { OFFICE_GROUP_IDS } from '~~/stubs/create-office-group-stub'
import { createFaker } from '~~/stubs/fake'
import { officeNames } from '~~/stubs/fake/office-names'
import { CreateStubs, ID_MIN, SEEDS } from '~~/stubs/index'

export const OFFICE_ID_MAX = ID_MIN + officeNames.length - 1
export const OFFICE_ID_MIN = ID_MIN
export const OFFICE_STUB_COUNT = OFFICE_ID_MAX - OFFICE_ID_MIN + 1

export function createOfficeStub (id: OfficeId = OFFICE_ID_MIN): Office {
  const { name, abbr, phoneticName } = officeNames[id - 1]
  const faker = createFaker(SEEDS[id - 1])
  const fake = faker.createFake()
  const purpose = id === OFFICE_ID_MIN ? Purpose.internal : faker.randomElement([Purpose.internal, Purpose.external])
  const dwsGenericService = faker.randomElement([{
    code: faker.randomNumericString(10),
    openedOn: faker.randomDateString(),
    designationExpiredOn: faker.randomDateString(),
    dwsAreaGradeId: faker.intBetween(DWS_AREA_GRADE_ID_MIN, DWS_AREA_GRADE_ID_MAX)
  }, undefined])
  const dwsCommAccompanyService = faker.randomElement([{
    code: faker.randomNumericString(10),
    openedOn: faker.randomDateString(),
    designationExpiredOn: faker.randomDateString()
  }, undefined])
  const ltcsHomeVisitLongTermCareService = faker.randomElement([{
    code: faker.randomNumericString(10),
    openedOn: faker.randomDateString(),
    designationExpiredOn: faker.randomDateString(),
    ltcsAreaGradeId: faker.intBetween(LTCS_AREA_GRADE_ID_MIN, LTCS_AREA_GRADE_ID_MAX)
  }, undefined])
  const ltcsCareManagementService = faker.randomElement([{
    code: faker.randomNumericString(10),
    openedOn: faker.randomDateString(),
    designationExpiredOn: faker.randomDateString(),
    ltcsAreaGradeId: faker.intBetween(LTCS_AREA_GRADE_ID_MIN, LTCS_AREA_GRADE_ID_MAX)
  }, undefined])
  const ltcsCompHomeVisitingService = faker.randomElement([{
    code: faker.randomNumericString(10),
    openedOn: faker.randomDateString(),
    designationExpiredOn: faker.randomDateString()
  }, undefined])
  const ltcsPreventionService = faker.randomElement([{
    code: faker.randomNumericString(10),
    openedOn: faker.randomDateString(),
    designationExpiredOn: faker.randomDateString()
  }, undefined])

  return {
    id,
    name,
    abbr,
    phoneticName: toKatakana(phoneticName),
    corporationName: Purpose.internal ? '' : name, // 仮
    phoneticCorporationName: Purpose.internal ? '' : toKatakana(phoneticName), // 仮
    purpose,
    addr: fake.addr,
    location: fake.location,
    tel: fake.tel,
    fax: fake.fax,
    email: fake.email,
    qualifications: faker.randomElements(OfficeQualification.values, faker.intBetween(0, OfficeQualification.size)),
    officeGroupId: purpose === Purpose.internal ? faker.randomElement(OFFICE_GROUP_IDS) : undefined,
    dwsGenericService,
    dwsCommAccompanyService,
    ltcsHomeVisitLongTermCareService,
    ltcsCareManagementService,
    ltcsCompHomeVisitingService,
    ltcsPreventionService,
    status: faker.randomElement(OfficeStatus.values),
    isEnabled: true,
    version: 1,
    createdAt: faker.randomDateTimeString(),
    updatedAt: faker.randomDateTimeString()
  }
}

export const createOfficeStubs: CreateStubs<Office> = (n = OFFICE_STUB_COUNT, skip = 0) => {
  const start = OFFICE_ID_MIN + skip
  const end = Math.min(start + n - 1, OFFICE_ID_MAX)
  return range(start, end).map(createOfficeStub)
}
