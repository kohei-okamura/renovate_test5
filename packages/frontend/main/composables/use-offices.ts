/*
 * Copyright Â© 2021 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { computed, reactive, toRefs, watch } from '@nuxtjs/composition-api'
import { OfficeQualification } from '@zinger/enums/lib/office-qualification'
import { Permission } from '@zinger/enums/lib/permission'
import { Purpose } from '@zinger/enums/lib/purpose'
import { usePlugins } from '~/composables/use-plugins'
import { OfficeId } from '~/models/office'
import { UserId } from '~/models/user'
import { VSelectOption } from '~/models/vuetify'
import { RefOrValues, unref } from '~/support/reactive'

type BaseParams = {
  isCommunityGeneralSupportCenter?: boolean
  permission?: Permission
  qualifications?: OfficeQualification[]
  userId?: UserId
}
type ParamsWithPurpose = BaseParams & {
  internal?: never
  purpose?: Exclude<Purpose, typeof Purpose.unknown>
}
type ParamsWithInternal = BaseParams & {
  internal: true
  purpose?: never
}
type Params = RefOrValues<ParamsWithPurpose | ParamsWithInternal>

export type OfficeOption = VSelectOption<OfficeId> & {
  keyword: string
}

const createOfficeResolver = (xs: VSelectOption<OfficeId>[]) => {
  return (id: OfficeId | undefined, alternative = '-') => (id && xs.find(x => x.value === id)?.text) ?? alternative
}

export const useOffices = (params: Params) => {
  const { $api } = usePlugins()
  const state = reactive({
    isLoadingOffices: true,
    officeOptions: [] as OfficeOption[]
  })
  const resolveOfficeAbbr = computed(() => createOfficeResolver(state.officeOptions))
  watch(
    computed(() => ({
      isCommunityGeneralSupportCenter: unref(params.isCommunityGeneralSupportCenter),
      internal: unref(params.internal),
      permission: unref(params.permission),
      purpose: unref(params.purpose),
      qualifications: unref(params.qualifications),
      userId: unref(params.userId)
    })),
    async ({ internal, purpose, ...rest }) => {
      state.isLoadingOffices = true
      state.officeOptions = await $api.options.offices({ ...rest, purpose: internal ? Purpose.internal : purpose })
      state.isLoadingOffices = false
    },
    { immediate: true }
  )
  return {
    ...toRefs(state),
    resolveOfficeAbbr
  }
}
