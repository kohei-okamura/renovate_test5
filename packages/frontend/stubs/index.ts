/*
 * Copyright Â© 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { createFaker } from '~~/stubs/fake'

export const ID_MAX = 100
export const ID_MIN = 1
export const STUB_COUNT = 1000
export const STUB_DEFAULT_SEED = 'zinger'

export const createSeeds = (n: number, seed: string = STUB_DEFAULT_SEED): string[] => {
  const faker = createFaker(seed)
  return Array.from(Array(n), () => faker.randomString(8))
}

export const SEEDS = createSeeds(STUB_COUNT * 3)

export type CreateStubs<T> = {
  (n?: number, skip?: number): T[]
}
