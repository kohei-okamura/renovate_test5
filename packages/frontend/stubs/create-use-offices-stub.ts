/*
 * Copyright Â© 2021 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { computed, reactive, toRefs } from '@nuxtjs/composition-api'
import { useOffices } from '~/composables/use-offices'
import { Office, OfficeId } from '~/models/office'
import { createOfficeStubs } from '~~/stubs/create-office-stub'

export const createUseOfficesStub = (offices?: Office[]): ReturnType<typeof useOffices> => {
  const options = (offices ?? createOfficeStubs()).map(x => ({ value: x.id, text: x.abbr, keyword: x.name }))
  const resolveOfficeAbbr = computed(() => (id: OfficeId | undefined, alternative = '-') => {
    return (id && options.find(x => x.value === id)?.text) ?? alternative
  })
  const data = reactive({
    isLoadingOffices: false,
    officeOptions: options
  })
  return {
    ...toRefs(data),
    resolveOfficeAbbr
  }
}
