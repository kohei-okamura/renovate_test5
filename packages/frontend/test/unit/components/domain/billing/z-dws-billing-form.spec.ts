/*
 * Copyright © 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { Wrapper } from '@vue/test-utils'
import { mocked } from '@zinger/helpers/testing/mocked'
import Vue from 'vue'
import ZDwsBillingForm from '~/components/domain/billing/z-dws-billing-form.vue'
import { useOffices } from '~/composables/use-offices'
import { DwsBillingsApi } from '~/services/api/dws-billings-api'
import { ValidationObserverInstance } from '~/support/validation/types'
import { OFFICE_ID_MIN } from '~~/stubs/create-office-stub'
import { createUseOfficesStub } from '~~/stubs/create-use-offices-stub'
import { createMockedFormService } from '~~/test/helpers/create-mocked-form-service'
import { getValidationObserver } from '~~/test/helpers/get-validation-observer'
import { setData } from '~~/test/helpers/set-data'
import { setupComponentTest } from '~~/test/helpers/setup-component-test'

jest.mock('~/composables/use-offices')

describe('z-dws-billing-form.vue', () => {
  const { mount } = setupComponentTest()
  const $form = createMockedFormService()
  const mocks = {
    $form
  }
  const form: DwsBillingsApi.CreateForm = {
    officeId: OFFICE_ID_MIN,
    transactedIn: '2020-10',
    providedIn: '2020-10'
  }
  const propsData = {
    errors: {},
    progress: false,
    value: form,
    buttonText: '作成開始'
  }

  let wrapper: Wrapper<Vue>

  function mountComponent () {
    wrapper = mount(ZDwsBillingForm, { propsData, mocks })
  }

  function unmountComponent () {
    wrapper.destroy()
  }

  beforeAll(() => {
    mocked(useOffices).mockReturnValue(createUseOfficesStub())
  })

  afterAll(() => {
    mocked(useOffices).mockReset()
  })

  it('should be rendered correctly', () => {
    mountComponent()
    expect(wrapper).toMatchSnapshot()
    unmountComponent()
  })

  describe('validation', () => {
    let observer: ValidationObserverInstance

    async function validate (values: Partial<DwsBillingsApi.CreateForm> = {}): Promise<void> {
      await setData(wrapper, {
        form: { ...form, ...values }
      })
      await observer.validate()
      jest.runOnlyPendingTimers()
    }

    beforeAll(() => {
      mountComponent()
      observer = getValidationObserver(wrapper)
    })

    afterAll(() => {
      unmountComponent()
    })

    it('should pass when input correctly', async () => {
      await validate()
      expect(observer).toBePassed()
    })

    it.each([
      ['officeId', 'office', { officeId: undefined }],
      ['transactedIn', 'transactedIn', { transactedIn: undefined }]
    ])('should fail when %s is empty', async (_, name, value) => {
      const messageWrapper = wrapper.find(`[data-${name}] .v-messages`)
      expect(messageWrapper.text()).toBe('')
      await validate(value)
      expect(observer).not.toBePassed()
      expect(messageWrapper.text()).toBe('入力してください。')
    })
  })
})
