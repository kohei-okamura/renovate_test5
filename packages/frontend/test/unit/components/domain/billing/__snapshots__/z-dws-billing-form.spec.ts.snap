// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`z-dws-billing-form.vue should be rendered correctly 1`] = `
<form
  buttontext="作成開始"
  class="v-form z-dws-billing-form"
  data-form=""
  novalidate="novalidate"
>
  <div>
    <div
      class="z-form-card-stub"
      title="基本情報"
    >
      <div
        class="z-form-card-stub-slot-actions"
      />
      <z-form-card-item-set-stub
        icon="M18,15H16V17H18M18,11H16V13H18M20,19H12V17H14V15H12V13H14V11H12V9H20M10,7H8V5H10M10,11H8V9H10M10,15H8V13H10M10,19H8V17H10M6,7H4V5H6M6,11H4V9H6M6,15H4V13H6M6,19H4V17H6M12,7V3H2V21H22V7H12Z"
        label=""
      >
        <div
          class="root"
          data-office=""
          data-z-form-card-item=""
        >
          <div
            class="v-input v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-select v-autocomplete"
            clearable=""
            error-messages=""
            label="事業所 *"
            value="1"
          >
            <div
              class="v-input__control"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                class="v-input__slot"
                role="combobox"
                style="background-color: rgb(241, 246, 250); border-color: #f1f6fa;"
              >
                <fieldset
                  aria-hidden="true"
                >
                  <legend
                    style="width: 0px;"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
                <div
                  class="v-select__slot"
                >
                  <label
                    class="v-label v-label--active theme--light"
                    style="left: 0px; position: absolute;"
                  >
                    事業所 *
                  </label>
                  <input
                    autocomplete="off"
                    type="text"
                  />
                  <div
                    class="v-input__append-inner"
                  >
                    <div
                      class="v-input__icon v-input__icon--clear"
                    >
                      <button
                        aria-label="clear icon"
                        class="v-icon notranslate v-icon--link mdi mdi-close theme--light"
                        type="button"
                      />
                    </div>
                  </div>
                  <div
                    class="v-input__append-inner"
                  >
                    <div
                      class="v-input__icon v-input__icon--append"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon notranslate mdi mdi-menu-down theme--light"
                      />
                    </div>
                  </div>
                  <input
                    type="hidden"
                    value="1"
                  />
                </div>
                <div
                  class="v-menu"
                >
                  <!---->
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <transition-group-stub
                    class="v-messages__wrapper"
                    name="message-transition"
                    tag="div"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </z-form-card-item-set-stub>
      <z-form-card-item-set-stub
        icon="M9,10V12H7V10H9M13,10V12H11V10H13M17,10V12H15V10H17M19,3A2,2 0 0,1 21,5V19A2,2 0 0,1 19,21H5C3.89,21 3,20.1 3,19V5A2,2 0 0,1 5,3H6V1H8V3H16V1H18V3H19M19,19V8H5V19H19M9,14V16H7V14H9M13,14V16H11V14H13M17,14V16H15V14H17Z"
        label=""
      >
        <div
          class="root"
          data-transactedin=""
          data-z-form-card-item=""
        >
          <div
            class="z-date-field"
            max="2018-05-17"
          >
            <div
              class="v-input v-input--is-label-active v-input--is-dirty v-input--is-readonly v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                  style="background-color: rgb(241, 246, 250); border-color: #f1f6fa;"
                >
                  <fieldset
                    aria-hidden="true"
                  >
                    <legend
                      style="width: 0px;"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active theme--light"
                      style="left: 0px; position: absolute;"
                    >
                      処理対象年月 *
                    </label>
                    <input
                      readonly="readonly"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <transition-group-stub
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
            <v-dialog-stub
              closedelay="0"
              contentclass=""
              data-z-date-field-dialog=""
              opendelay="0"
              origin="center center"
              retainfocus="true"
              transition="dialog-transition"
              width="290px"
            >
              <v-date-picker-stub
                data-z-date-field-picker=""
                dayformat="[Function]"
                eventcolor="warning"
                firstdayofweek="1"
                headerdateformat="[Function]"
                locale="ja-JP"
                localefirstdayofyear="0"
                max="2018-05-17"
                min="1920-01-01"
                nexticon="$next"
                nextmontharialabel="$vuetify.datePicker.nextMonthAriaLabel"
                nextyeararialabel="$vuetify.datePicker.nextYearAriaLabel"
                notitle="true"
                previcon="$prev"
                prevmontharialabel="$vuetify.datePicker.prevMonthAriaLabel"
                prevyeararialabel="$vuetify.datePicker.prevYearAriaLabel"
                selecteditemstext="$vuetify.datePicker.itemsSelected"
                showcurrent="true"
                titledateformat="[Function]"
                type="month"
                value="2020-10-01"
                width="290"
                yearformat="[Function]"
              >
                <v-spacer-stub
                  tag="div"
                />
                <a
                  class="v-btn-stub"
                  color="primary"
                  data-z-date-field-button-clear=""
                  text=""
                >
                  クリア
                </a>
                <a
                  class="v-btn-stub"
                  color="primary"
                  data-z-date-field-button-cancel=""
                  text=""
                >
                  キャンセル
                </a>
              </v-date-picker-stub>
            </v-dialog-stub>
          </div>
        </div>
      </z-form-card-item-set-stub>
      <div
        class="z-form-card-stub-slot-header"
      />
    </div>
    <z-form-action-button-stub
      icon="M14,13V17H10V13H7L12,8L17,13M19.35,10.03C18.67,6.59 15.64,4 12,4C9.11,4 6.6,5.64 5.35,8.03C2.34,8.36 0,10.9 0,14A6,6 0 0,0 6,20H19A5,5 0 0,0 24,15C24,12.36 21.95,10.22 19.35,10.03Z"
      text="作成開始"
    />
  </div>
</form>
`;
